spirv_glsl.cpp

```diff
@@ -6467,7 +6467,10 @@ void CompilerGLSL::emit_unary_func_op(uint32_t result_type, uint32_t result_id,
 void CompilerGLSL::emit_binary_func_op(uint32_t result_type, uint32_t result_id, uint32_t op0, uint32_t op1,
                                        const char *op)
 {
-	bool forward = should_forward(op0) && should_forward(op1);
+	// Sampled image types must always be forwarded in GLSL unless they are a uniform variable
+	const auto &type = get_type(result_type);
+	bool must_forward = type.basetype == SPIRType::SampledImage && type.storage != spv::StorageClassUniform;
```

Comment:
This check is not complete. Need to consider Sampler and Image as well. There should be a helper somewhere for "is_opaque_type" or something like that.

Reply:
Thanks for the hint, I added `type_is_opaque_value` to the condition.